src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -2727,18 +2759,13 @@ bool BBPredsChecker::CheckEHFinallyRet(BasicBlock* blockPred, BasicBlock* block)
     // block.  We can shorten the search by only looking in places where it is legal
     // to have a call to the finally.
 
-    BasicBlock* begBlk;
-    BasicBlock* endBlk;
-    comp->ehGetCallFinallyBlockRange(hndIndex, &begBlk, &endBlk);
+    BasicBlock* firstBlock;
+    BasicBlock* lastBlock;
+    comp->ehGetCallFinallyBlockRange(hndIndex, &firstBlock, &lastBlock);
 
-    for (BasicBlock* bcall = begBlk; bcall != endBlk; bcall = bcall->Next())
+    for (BasicBlock* const bcall : comp->Blocks(firstBlock, lastBlock))
     {
-        if (!bcall->KindIs(BBJ_CALLFINALLY) || !bcall->HasJumpTo(finBeg))
-        {
-            continue;
-        }
-
-        if (bcall->NextIs(block))
+        if (bcall->KindIs(BBJ_CALLFINALLY) && bcall->HasJumpTo(finBeg) && bcall->NextIs(block))
```

Comment:
From what I can tell this doesn't actually check that `block` is in the ehf successor table for `blockPred` -- should we check that too?

Reply:
Yes, will do.